### ISSUE TITLE
[match] Fails to checkout master branch if it is not the default branch of the repository

### ISSUE THREAD

**Issue by mbogh** [fastlane/fastlane#16475]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I changed the default branch of my certificates repository to something different than `master` and now **match** fails to sync any certificate and/or profile on the `master` branch.

If ` unless self.branch == "master"` is removed from [git_storage.rb](https://github.com/fastlane/fastlane/blob/master/match/lib/match/storage/git_storage.rb#L116) it works on both repositories with `master` as a default branch and not.

So the solution could either be to always checkout the specified branch or check the working directory if there is a branch mismatch.

##### Command executed

```
sync_code_signing(
      git_url: "git@github.com:ORG/certificates.git",
      git_branch: 'master',
      app_identifier: ['com.abc.xyz'],
      type: 'appstore',
      team_id: "XXX",
      readonly: true
    )
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> 
[✔] 🚀 
[08:52:13]: ------------------------------
[08:52:13]: --- Step: default_platform ---
[08:52:13]: ------------------------------
[08:52:13]: Driving the lane 'ios app' 🚀
[08:52:13]: -------------------------------
[08:52:13]: --- Step: sync_code_signing ---
[08:52:13]: -------------------------------

+----------------------------+--------------------------------------------+
|                        Summary for match 2.148.1                        |
+----------------------------+--------------------------------------------+
| git_url                    | git@github.com:ORG/certificates.git        |
| git_branch                 | master                                     |
| app_identifier             | ["com.abc.xyz"]                            |
| type                       | appstore                                   |
| team_id                    | xxx                                        |
| readonly                   | true                                       |
| git_private_key            | ********                                   |
| generate_apple_certs       | true                                       |
| skip_provisioning_profiles | false                                      |
| storage_mode               | git                                        |
| shallow_clone              | false                                      |
| clone_branch_directly      | false                                      |
| keychain_name              | login.keychain                             |
| force                      | false                                      |
| force_for_new_devices      | false                                      |
| skip_confirmation          | false                                      |
| skip_docs                  | false                                      |
| platform                   | ios                                        |
| fail_on_name_taken         | false                                      |
| verbose                    | false                                      |
+----------------------------+--------------------------------------------+

[08:52:13]: Private key file does not exist, will continue by using it as a raw key.
[08:52:13]: Cloning remote git repo...
[08:52:13]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[08:52:18]: 🔓  Successfully decrypted certificates repo
[08:52:18]: Couldn't find a valid code signing identity for distribution... creating one for you now
+------------------+---------+
|        Lane Context        |
+------------------+---------+
| DEFAULT_PLATFORM | ios     |
| PLATFORM_NAME    | ios     |
| LANE_NAME        | ios app |
+------------------+---------+
[08:52:18]: No code signing identity found and can not create a new one because you enabled `readonly`

+------+-------------------+-------------+
|            fastlane summary            |
+------+-------------------+-------------+
| Step | Action            | Time (in s) |
+------+-------------------+-------------+
| 1    | default_platform  | 0           |
| 💥   | sync_code_signing | 4           |
+------+-------------------+-------------+

[08:52:18]: fastlane finished with errors

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/fastlane)
FastlaneCore::Interface::FastlaneCrash: [!] No code signing identity found and can not create a new one because you enabled `readonly`
  /Users/user/Private/github/fastlane/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!'
  /Users/user/Private/github/fastlane/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
  /Users/user/Private/github/fastlane/match/lib/match/runner.rb:159:in `fetch_certificate'
  /Users/user/Private/github/fastlane/match/lib/match/runner.rb:93:in `run'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/actions/sync_code_signing.rb:13:in `run'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:261:in `block (2 levels) in execute_action'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:253:in `block in execute_action'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
  Fastfile:20:in `block (2 levels) in parsing_binding'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/user/Private/github/fastlane/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/user/Private/github/fastlane/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  /Users/user/Private/github/fastlane/bin/fastlane:23:in `<top (required)>'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/fastlane:23:in `load'
  /Users/user/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/fastlane:23:in `<top (required)>'
</pre>
</details>

### Environment

<details>
  <pre> 
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                    |
| --------------------------- | -------------------------------------------------------- |
| OS                          | 10.15.4                                                  |
| Ruby                        | 2.6.5                                                    |
| Bundler?                    | true                                                     |
| Git                         | git version 2.24.1 (Apple Git-126)                       |
| Installation Source         | ~/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/fastlane |
| Host                        | Mac OS X 10.15.4 (19E287)                                |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.5/lib                              |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                              |
| Is contained                | false                                                    |
| Is homebrew                 | false                                                    |
| Is installed via Fabric.app | false                                                    |
| Xcode Path                  | /Applications/Xcode-11.4.app/Contents/Developer/         |
| Xcode Version               | 11.4                                                     |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     |             |   |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.148.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| bundler                   | 2.1.4        |
| psych                     | 3.1.0        |
| stringio                  | 0.0.2        |
| strscan                   | 1.0.0        |
| rake                      | 11.3.0       |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| ast                       | 2.4.0        |
| atomos                    | 0.1.3        |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.314.0      |
| aws-sigv4                 | 1.1.3        |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.95.0       |
| aws-sdk-kms               | 1.31.0       |
| aws-sdk-s3                | 1.64.0       |
| babosa                    | 1.0.3        |
| debug_inspector           | 0.0.3        |
| binding_of_caller         | 0.8.0        |
| builder                   | 3.2.4        |
| byebug                    | 11.1.3       |
| claide                    | 1.0.3        |
| climate_control           | 0.2.0        |
| coderay                   | 1.1.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| json                      | 2.3.0        |
| docile                    | 1.3.2        |
| simplecov-html            | 0.10.2       |
| simplecov                 | 0.16.1       |
| sync                      | 0.5.0        |
| tins                      | 1.25.0       |
| term-ansicolor            | 1.7.1        |
| thor                      | 1.0.1        |
| coveralls                 | 0.8.23       |
| safe_yaml                 | 1.0.5        |
| crack                     | 0.4.3        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| diff-lcs                  | 1.3          |
| digest-crc                | 0.5.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.5        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.73.0       |
| fakefs                    | 0.8.1        |
| multipart-post            | 2.0.0        |
| faraday                   | 1.0.1        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.2       |
| multi_json                | 1.14.1       |
| os                        | 1.1.0        |
| signet                    | 0.14.0       |
| googleauth                | 0.12.0       |
| httpclient                | 2.8.3        |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.26.1       |
| mini_magick               | 4.10.1       |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.16.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| hashdiff                  | 0.3.9        |
| http-accept               | 1.7.0        |
| interception              | 0.5          |
| method_source             | 1.0.0        |
| mime-types-data           | 3.2020.0512  |
| mime-types                | 3.3.1        |
| netrc                     | 0.11.0       |
| parallel                  | 1.19.1       |
| parser                    | 2.7.1.2      |
| powerpack                 | 0.1.2        |
| pry                       | 0.13.1       |
| pry-byebug                | 3.9.0        |
| pry-rescue                | 1.5.0        |
| pry-stack_explorer        | 0.4.9.3      |
| rack                      | 1.6.13       |
| rack-protection           | 1.5.5        |
| rainbow                   | 2.2.2        |
| rb-readline               | 0.5.5        |
| rest-client               | 2.1.0        |
| rspec-support             | 3.5.0        |
| rspec-core                | 3.5.4        |
| rspec-expectations        | 3.5.0        |
| rspec-mocks               | 3.5.0        |
| rspec                     | 3.5.0        |
| rspec_junit_formatter     | 0.2.3        |
| ruby-progressbar          | 1.10.1       |
| rubocop                   | 0.49.1       |
| rubocop-require_tools     | 0.1.2        |
| tilt                      | 2.0.10       |
| sinatra                   | 1.4.8        |
| webmock                   | 2.3.2        |
| xcov                      | 1.4.3        |
| yard                      | 0.9.25       |
</details>

*generated on:* **2020-05-19**
</details>
</pre>
</details>

**Comment by mbogh** [comment]
If there is a preferred solution, I will be happy to implement it and make pull request 💯

**Comment by mbogh** [comment]
@janpio I would love to fix this in a pull request, just need a direction to go with 🙅

**Comment by janpio** [comment]
What could the negative effects be of always checking out the branch, even if it's master already?

**Comment by mbogh** [comment]
I honestly do not know, when it is done you would just get `Already on 'branch-name'` even if there are on stages/commited changes.

**Comment by janpio** [comment]
Then I think we are good for that PR and will see if anyone has reservations about merging it, otherwise we do and it is in the next release.

**Comment by mbogh** [comment]
This was fixed by PR #16622 
/close
